src/main/java/algorithms/medium/BestTimeToBuyAndSellStockWithCooldown.java

```diff
@@ -0,0 +1,24 @@
+package algorithms.medium;
+
+public class BestTimeToBuyAndSellStockWithCooldown {
+
+    public int maxProfit(int[] prices) {
+
+        int totalProfit = 0, restProfit = 0, currBuyCosts = Integer.MIN_VALUE;
```

Comment:
@altayhunoglu Do you know a better naming for the variable `currBuyCosts`. From finance, I can only think of "deficit" and "loss". But these only describe states where total profit is less than 0, not when are total profit goes below its current state because we have spent it on something, we might be still profiting here when we buy a stock, even before we sell.